File: zb1/codingTest/220317/MostAppearNumber/MostAppearNumber.js
/**
 * @param s {string}
 * @return {number}
 */
function solution(s) {
  const numsCount = new Array(10).fill(0);

  for (let i = 0; i < s.length; i++) {
    const c = s.charAt(i);
    const num = Number(c);

    numsCount[num] += 1;
  }

  return mostAppearNum(numsCount);
}

function mostAppearNum(numsCount) {
  let ret = -1;
  let cnt = 0;

  for (let i = 0; i < 10; i++) {
    if (cnt < numsCount[i]) {
      ret = i;
      cnt = numsCount[i];
    }
  }

  return ret;
}

solution

File: zb1/codingTest/220317/o/(1)MostAppearNumber/MostAppearNumber.py
import math
from random import random


def solution(s):
    """
    :param s: str
    :return: int
    """

    numsCount = [0] * 10

    for i in range(len(s)):
        c = s[i]
        num = int(c)

        numsCount[num] += 1

    return mostAppearNum(numsCount)


def mostAppearNum(numsCount):
    result = -1
    cnt = 0

    for i in range(10):
        if cnt < numsCount[i]:
            result = i
            cnt = numsCount[i]

    return result
